plugins/scripts/upload/trns_transform_Genbank_Genome_to_KBaseGenomeAnnotations_GenomeAnnotation.py

```diff
@@ -1677,23 +1683,86 @@ def upload_genome(shock_service_url=None,
+            #1073741824 bytes is a GB.  This gives ~7%cushion for potential unicode 16 characters
+            if container_byte_estimate > 1000000000 : 
+                #Create an ordered dict of features and lengths
+                temp_dict = dict()
+                for feature in feature_container['features']:
+                    temp_dict[feature] = feature_container['features'][feature]['dna_sequence_length']
+                container_feature_lengths = OrderedDict(sorted(temp_dict.items(), key=lambda t: t[1]))
+            while container_byte_estimate > 1000000000 :
+                remove_count = 10
+                container_length = len(container_feature_lengths)
+                if container_length < 10 :
+                    remove_count = container_length
+                if container_length == 0:
+                    #IF CORE TYPE THROW AN ERROR
+                    if feature_container['type'] in ['gene','CDS','mRNA']:
+                        raise Exception("The resulting %s feature type container is above the workspace 1GB limit." % (feature_container['type']))
+                    else:
+                        #REMOVE FEATURE TYPE AND CLEANUP ALIASES (IF NOT CORE FEATURE TYPE) AND DO WARNING  (DONT SAVE CONTAINER)
+                        do_feature_container_save = False
+                        for alias in feature_lookup_dict:
+                            f_counter = 0
+                            for temp_fc_ref, temp_fc_id in feature_lookup_dict[alias]:
+                                if feature_container['type'] == reverse_feature_container_ref_lookup[temp_fc_ref]:
+                                    #HOW TO REMOVE TUPLE FROM LIST IN PLACE?
+                                    del( feature_lookup_dict[alias][f_counter])
+                                else:
+                                    # important because the positions of all values will be decremented on a delete
+                                    # so only increment if you did not delete                   
+                                    f_counter += 1
+                            if len(feature_lookup_dict[alias]) == 0:
+                                #Need to remove the alias.
+                                del( feature_lookup_dict[alias]) 
+                        annotation_quality_warnings.append("The feature type {} was unable to be saved due to workspace size limitations".format(feature_container['type']))
+                        break
+
+                #remove 10 sequences at a time and retry (or what's left)
+                for i in range(container_length):
+                    removed_sequence_feature_id = container_feature_lengths.popitem()[0]
+                    features_with_sequences_removed_list.append(removed_sequence_feature_id)
+                    feature_container['features'][removed_sequence_feature_id]['dna_sequence']=""
+                container_byte_estimate = len(simplejson.dumps(feature_container))
```

Comment:
see comment about this above. Here you could reuse the suggested calculate_feature_bytes for the removed feature and simply subtract that from the total.